narwhals/translate.py

```diff
@@ -254,7 +255,7 @@ def from_native(
 
 @overload
 def from_native(
-    native_object: Any,
+    native_object: IntoSeriesT,
```

Comment:
Due to the following code in `altair`, should we keep the `Any` type and introduce a new one specifically for `IntoSeriesT`? @MarcoGorelli

```python
def _from_array_like(obj: Iterable[Any], /) -> list[Any]:
    try:
        ser = nw.from_native(obj, strict=True, series_only=True)
        return ser.to_list()
    except TypeError:
        return list(obj)
```
from https://github.com/vega/altair/blob/11078c2676a921cdfc44477ffe0d2a43609e0306/altair/utils/schemapi.py#L502-L507